### PR TITLE
Bookmark question from sidebar

### PR DESCRIPTION
**Pull request by gusaiani**
Resolves: https://github.com/metabase/metabase/issues/20322

Step 1 of the bookmarks feature.

### How to Test

1. Open a question page (can be a model page too)
2. Open sidebar
3. Click on the ribbon icon

image

#### We expect this to happen

Ribbon should toggle between blue and black.
Blue: is bookmarked.
Black: is _not_ bookmarked.

Network tab should show POST and DELETE bookmark requests on click.

##### Notes

This iteration does not retrieve `isBookmarked` boolean from server (coming soon).

Hence, if you bookmark a question, reload its page and bookmark it, the server will error: you are bookmarking a bookmarked question. Better error treatment to come.

### CODE REVIEW

**Comment by mazameli** [comment]
> Hence, if you bookmark a question, reload its page and bookmark it, the server will error: you are bookmarking a bookmarked question.

Also appears that if I bookmark a question then reload or come back to it later, the icon is black again instead of staying blue as expected.

**Comment by gusaiani** [comment]
> > Hence, if you bookmark a question, reload its page and bookmark it, the server will error: you are bookmarking a bookmarked question.
> 
> Also appears that if I bookmark a question then reload or come back to it later, the icon is black again instead of staying blue as expected.

Back-end is not ready yet. It's not saving at the moment :)

**Review by daltojohnso — APPROVED** [review]
👍

File: frontend/src/metabase/query_builder/components/QuestionActionButtons.jsx [hunk 1]
```diff
@@ -127,6 +135,15 @@ function QuestionActionButtons({
           />
         </Tooltip>
       )}
+      <Tooltip tooltip={t`Bookmark`}>
+        <Button
+          onlyIcon
+          icon="bookmark"
+          iconSize={ICON_SIZE}
+          onClick={toggleBookmark}
+          style={{ color: bookmarkButtonColor }}
```
[the remark this answers is not in the document]
Comment (gusaiani) [reply frontend/src/metabase/query_builder/components/QuestionActionButtons.jsx]: Thank you Dalton.

**Pull request merged**